.github/workflows/main.yml

```diff
@@ -0,0 +1,147 @@
+        tar -xvf kubebuilder_${{env.KUBEBUILDER_VERSION}}_linux_amd64.tar.gz
+        mv kubebuilder_${{env.KUBEBUILDER_VERSION}}_linux_amd64 /usr/local/kubebuilder
+
+    - name: Vet and Build
+      run: make manager
+
+  test:
+    name: Test
+    container: 
+      image: golang:1.15
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: Get dependencies
+      run: |
+        go get -v -t -d ./...
+        
+    - name: Add kubebuilder
+      run:  |
+        curl -L https://github.com/kubernetes-sigs/kubebuilder/releases/download/v${{env.KUBEBUILDER_VERSION}}/kubebuilder_${{env.KUBEBUILDER_VERSION}}_linux_amd64.tar.gz > kubebuilder_${{env.KUBEBUILDER_VERSION}}_linux_amd64.tar.gz
+        tar -xvf kubebuilder_${{env.KUBEBUILDER_VERSION}}_linux_amd64.tar.gz
+        mv kubebuilder_${{env.KUBEBUILDER_VERSION}}_linux_amd64 /usr/local/kubebuilder
+
+    - name: Test
+      run: make test
+
+    - name: Coverage
+      uses: codecov/codecov-action@v1
+      with:
+        # token: ${{ secrets.CODECOV_TOKEN }} # not required for public repos
+        file: ./cover.out
+        # flags: unittests # optional
+        name: externalsecret-operator
+        fail_ci_if_error: true 
+
+  docker:
+    name: Docker
+    runs-on: ubuntu-latest
+    needs: [build, test]
+    steps:
+    - name: Prepare
+      id: prep
+      run: |
+        DOCKER_IMAGE=ghcr.io/external-secrets/external-secrets
+
+        VERSION=edge
+        if [[ $GITHUB_REF == refs/tags/* ]]; then
+          VERSION=${GITHUB_REF#refs/tags/}
+        elif [[ $GITHUB_REF == refs/heads/* ]]; then
+          VERSION=$(echo ${GITHUB_REF#refs/heads/} | sed -r 's#/+#-#g')
+        elif [[ $GITHUB_REF == refs/pull/* ]]; then
+          VERSION=pr-${{ github.event.number }}
+        fi
+
+        TAGS="${DOCKER_IMAGE}:${VERSION}"
+        if [ "${{ github.event_name }}" = "push" ]; then
+          TAGS="$TAGS,${DOCKER_IMAGE}:sha-${GITHUB_SHA::8}"
+        fi
+
+        PUSH_IMAGE=true
+        REPO_FULL_NAME="${{ github.event.pull_request.head.repo.full_name }}"
+        # If this is both a pull request and a fork, then don't push the image
+        if [[ ${{ github.event_name }} == pull_request ]]; then
+          if [[ $REPO_FULL_NAME != external-secrets/external-secrets ]]; then
+            PUSH_IMAGE=false
+          fi
+        fi
+
+        echo ::set-output name=version::${VERSION}
+        echo ::set-output name=tags::${TAGS}
+        echo ::set-output name=created::$(date -u +'%Y-%m-%dT%H:%M:%SZ')
+        echo ::set-output name=push_image::$PUSH_IMAGE
+
+    - name: Check out the repo
+      uses: actions/checkout@v2
+      
+    - name: Set up QEMU
+      id: qemu
+      uses: docker/setup-qemu-action@v1
+      with:
+        platforms: all
```

Comment:
Is QEMU needed if we only build for `linux/amd64` ? Otherwise we should probably add more platforms or we can change all here to `linux/amd64`

Reply:
:thinking: I left the QEMU step there because we probably will go the multiplatform route later (right?), but I did not want to complicate the Dockerfile right now in the beginning. Limited to  `linux/amd64` for now.